pkg/util.go

```diff
@@ -71,15 +79,75 @@ func must(v []byte, err error) string {
 	return string(v)
 }
 
-func waitForDBReady(appBinding *v1alpha1.AppBinding, secret *core.Secret, waitTimeout int32) error {
+func (opt *postgresOptions) waitForDBReady(appBinding *v1alpha1.AppBinding, secret *core.Secret, waitTimeout int32) error {
```

Comment:
I feel like there should be a `createClient()` type function that can the reused across all three functions.